### DISCUSSION TITLE
Mime types / Target Folder

### DISCUSSION THREAD

**Discussion by Gekko23** [cwilby/nextcloud-workflow-media-converter#487] [General]
When using FLOW->CONVERT MEDIA using a _when [file is created] and [mime-type] [equals] [custom mime type]_ any regular expression i tried was rejected. I have no idea how to detect let's say any *.webm file or any .mov file.

Could someone please provide a list of video-mimetypes in the format that the flow system in Nextcloud expects? 

There is literally NO EXAMPLE anywhere to be found, not even in the documentation of Nextcloud itself. When using the nextcloud-workflow-media-converter, the flow system offers the option to customly select files of a certain mime type: what are the EXACT valid arguments here and how are they written EXACTLY?

**Comment by cwilby** [comment]
Hey @Gekko23!

Yeah, this one bugs me as well.  There's an open issue that I've been meaning to convert to a pull request to add file types as a dropdown to the flow engine. https://github.com/nextcloud/server/issues/27591. This is a good reminder to look into that, thanks.

Can you confirm that entering `/mov/` or `/webm/` in the regex match field instead of `*.mov` and `*.webm` works/doesn't work please?

**Reply by JeCheeseSmith** [reply]
Hi @cwilby , I recently got to know about this app and I was configuring it to convert some audio files when uploaded (very similar scenario as above).

For me it seems to work for .wav files when I fill in `/wav/`  the Mimetype. (Other inputs are rejected with:
> The given regular expression is invalid

I did catch the following error in the logs but the file was converted succesfully: 

> Error	workflow_media_converter	
> [OCP\Files\NotFoundException] :: (0) :: /admin/files/Documents/Music/test.wav :: #0 /var/www/html/lib/private/Files/Node/LazyFolder.php(161): OC\Files\Node\Root->get('/admin/files/Do...') #1 /var/www/html/custom_apps/workflow_media_converter/lib/BackgroundJobs/ConvertMediaJob.php(109): OC\Files\Node\LazyFolder->get('/admin/files/Do...') #2 /var/www/html/custom_apps/workflow_media_converter/lib/BackgroundJobs/ConvertMediaJob.php(65): OCA\WorkflowMediaConverter\BackgroundJobs\ConvertMediaJob->parseArguments(Array) #3 /var/www/html/lib/public/BackgroundJob/Job.php(80): OCA\WorkflowMediaConverter\BackgroundJobs\ConvertMediaJob->run(Array) #4 /var/www/html/lib/public/BackgroundJob/QueuedJob.php(61): OCP\BackgroundJob\Job->start(Object(OC\BackgroundJob\JobList)) #5 /var/www/html/lib/public/BackgroundJob/QueuedJob.php(47): OCP\BackgroundJob\QueuedJob->start(Object(OC\BackgroundJob\JobList)) #6 /var/www/html/cron.php(177): OCP\BackgroundJob\QueuedJob->execute(Object(OC\BackgroundJob\JobList)) #7 {main}

Yet i'm happy it works! Thanks for making/maintaining!

Would it be handy to add this to the ReadMe in the meantime if the issue doesn't progress further?
